pkg/nuctl/command/common/importreport.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2023 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/nuclio/logger"
+)
+
+type ProjectReports struct {
+	projectReports map[string]*ProjectReport
+}
+
+func NewProjectsReport() *ProjectReports {
+	return &ProjectReports{
+		projectReports: make(map[string]*ProjectReport),
+	}
+}
+
+func (psr *ProjectReports) AddReport(report *ProjectReport) {
+	psr.projectReports[report.Name] = report
+}
+
+func (psr *ProjectReports) GetOrCreate(projectName string) *ProjectReport {
+	report, found := psr.projectReports[projectName]
+	if !found {
+		report = NewProjectReport(projectName)
+		psr.AddReport(report)
+	}
+	return report
+}
+
+func (psr *ProjectReports) SaveToFile(ctx context.Context, loggerInstance logger.Logger, path string) {
+	saveReportToFile(ctx, loggerInstance, psr.projectReports, path)
+}
```

Comment:
To minimize duplications, this can be moved to an `AbstractReport` struct that the rest of the reports will "inherit".
It might also be useful in the future for adding common functionalities